File: PokemonGrabber/PokemonGrabber/Move.cs
using Newtonsoft.Json;

namespace PokemonGrabber;

public class Move
{
    [JsonProperty("id")]
    public int Id { get; set; }
    
    [JsonProperty("name")]
    public string Name { get; set; }
    
    [JsonProperty("type")]
    public string Type { get; set; }
    
    [JsonProperty("description")]
    public string Description { get; set; }
    
    [JsonProperty("power")]
    public int? Power { get; set; }
    
    [JsonProperty("accuracy")]
    public int? Accuracy { get; set; }
    
    [JsonProperty("pp")]
    public int PP { get; set; }
    
    [JsonProperty("priority")]
    public int Priority { get; set; }
    
    [JsonProperty("damage_class")]
    public string DamageClass { get; set; }
    
    [JsonProperty("target")]
    public string Target { get; set; }
}

File: Pokedex/Pokedex/Models/Move.swift
//
//  Move.swift
//  Pokedex
//
//  Created by Thomas Bernard on 23/07/2023.
//

import Foundation

struct Move : Decodable, Identifiable {
    let id : Int
    let name : String
    let type : Type
    let description : String
    let power : Int?
    let accuracy : Int?
    let pp : Int
    let priority : Int
    let damageClass : String
    let target : String
    
    enum CodingKeys: String, CodingKey {
        case id
        case name
        case type
        case description
        case power
        case accuracy
        case pp
        case priority
        case damageClass = "damage_class"
        case target
    }
}
